Makefile

```diff
@@ -0,0 +1,72 @@
+# Heavily inspired by:
+# - Lighthouse: https://github.com/sigp/lighthouse/blob/693886b94176faa4cb450f024696cb69cda2fe58/Makefile
+# - Reth: https://github.com/paradigmxyz/reth/blob/1f642353ca083b374851ab355b5d80207b36445c/Makefile
+.DEFAULT_GOAL := help
+
+# Cargo profile for builds.
+PROFILE ?= release
```

Comment:
This should not default to release for 99% of the use cases of this file

Reply:
Should we change it to `dev`?

https://doc.rust-lang.org/cargo/reference/profiles.html#dev

Users can indeed override this with `PROFILE=release make build` if they need a release build